pkg/auditlogs/client.go

```diff
@@ -0,0 +1,280 @@
+
+	// Optional list of targets to filter
+	Targets []string `json:"targets,omitempty"`
+}
+
+type CreateExportResponse struct {
+	Object    string `json:"object"`
+	Id        string `json:"id"`
+	State     string `json:"state"`
+	Url       string `json:"url"`
+	CreatedAt string `json:"created_at"`
+	UpdatedAt string `json:"updated_at"`
+}
+
+type GetExportOpts struct {
+	ExportId string `json:"export_id" binding:"required"`
+}
+
+type GetExportResponse struct {
+	Object    string `json:"object"`
+	Id        string `json:"id"`
+	State     string `json:"state"`
+	Url       string `json:"url"`
+	CreatedAt string `json:"created_at"`
+	UpdatedAt string `json:"updated_at"`
+}
+
+func (c *Client) init() {
+	if c.HTTPClient == nil {
+		c.HTTPClient = &http.Client{Timeout: 10 * time.Second}
+	}
+
+	if c.EventsEndpoint == "" {
+		c.EventsEndpoint = "https://api.workos.com/audit_logs/events"
+	}
+
+	if c.ExportsEndpoint == "" {
+		c.ExportsEndpoint = "https://api.workos.com/audit_logs/exports"
+	}
+
+	if c.JSONEncode == nil {
+		c.JSONEncode = json.Marshal
+	}
+}
+
+// CreateEvent creates an Audit Log event.
+func (c *Client) CreateEvent(ctx context.Context, e AuditLogEventOpts) error {
+	c.once.Do(c.init)
+
+	e.Event.OccurredAt = defaultTime(e.Event.OccurredAt)
+
+	data, err := c.JSONEncode(e)
+	if err != nil {
+		return err
+	}
+
+	req, err := http.NewRequest(http.MethodPost, c.EventsEndpoint, bytes.NewBuffer(data))
+	if err != nil {
+		return err
+	}
+	req = req.WithContext(ctx)
+	req.Header.Set("Content-Type", "application/json")
+	req.Header.Set("Authorization", "Bearer "+c.APIKey)
+	req.Header.Set("User-Agent", "workos-go/"+workos.Version)
+
+	if e.IdempotencyKey != "" {
+		req.Header.Set("Idempotency-Key", e.IdempotencyKey)
+	}
+
+	res, err := c.HTTPClient.Do(req)
+	if err != nil {
+		return err
+	}
+	defer res.Body.Close()
+
+	return workos_errors.TryGetHTTPError(res)
+}
+
+// CreateExport creates an export of Audit Log events. You can specify some filters.
+func (c *Client) CreateExport(ctx context.Context, e CreateExportOpts) (CreateExportResponse, error) {
+	c.once.Do(c.init)
+
+	data, err := c.JSONEncode(e)
+	if err != nil {
+		return CreateExportResponse{}, err
+	}
+
+	req, err := http.NewRequest(http.MethodPost, c.ExportsEndpoint, bytes.NewBuffer(data))
+	if err != nil {
+		return CreateExportResponse{}, err
+	}
+	req = req.WithContext(ctx)
+	req.Header.Set("Content-Type", "application/json")
+	req.Header.Set("Authorization", "Bearer "+c.APIKey)
+	req.Header.Set("User-Agent", "workos-go/"+workos.Version)
+
+	res, err := c.HTTPClient.Do(req)
+	if err != nil {
+		return CreateExportResponse{}, err
+	}
+	defer res.Body.Close()
+
+	if err = workos_errors.TryGetHTTPError(res); err != nil {
+		return CreateExportResponse{}, err
+	}
+
+	var body CreateExportResponse
+	dec := json.NewDecoder(res.Body)
+	err = dec.Decode(&body)
+	return body, err
+}
+
+// GetExport creates an export of Audit Log events. You can specify some filters.
```

Comment:
Is there a spot where I'd be able to see what is filtered? (Or should this comment be eventually updated with a not yet published doc?)

Reply:
Oh... That's a missing copy/paste. I'll update it, but it will eventually be updated after we publish the docs. 😄